### DISCUSSION TITLE
谁不想要 2^64 个 IP 的代理池 ？

### DISCUSSION THREAD

**Discussion by giscus** [zu1k/blog#53] [Comments]
# 谁不想要 2^64 个 IP 的代理池 ？

哪个小伙子不想要一个拥有 1844 6744 0737 0955 1616 个 IP 的代理池？本文中我将介绍一种充分利用 IPv6 资源来构建代理池的方法

https://zu1k.com/posts/tutorials/http-proxy-ipv6-pool/

**Comment by dp0d** [comment]
送了1000个，现在还是白嫖状态

**Comment by elizabeth-wilona** [comment]
只是允许绑定非本地存在IP,因为整个段可以用,所以等于绑定到任意,但是由于大量NDP报文,有时候会当作是一种flood.

**Reply by zu1k** [reply]
对，不知道ISP会不会管这些NDP报文

**Comment by sorz** [comment]
HE 的 tunnel broker 现在还开放注册吧？给 /48 的，而且是 route 过来的，没有什么 NDP 的问题（

**Reply by zu1k** [reply]
这样可有的玩了！

**Comment by ghost** [comment]
事实上大部分网站的IPv6限流都是根据整个/48段来的，例如 [Let's Encrypt](https://letsencrypt.org/docs/rate-limits/)

> You can create a maximum of 10 Accounts per IP Address per 3 hours. You can create a maximum of 500 Accounts per IP Range within an IPv6 /48 per 3 hours.

Google/Facebook也是这样，因此Linode等IDC的IPv6地址访问Google要求人机验证的概率在未登录账号的情况接近100%（以前测试是这样），而IPv4地址则很少出现人机验证。

**Reply by zu1k** [reply]
感谢告知，看来IPv6因为数量太大，信任度非常低啊

**Comment by yllhwa** [comment]
前些日子刚好用openresty的bind_ip实现了，有些网站的waf还是有问题的，偷着乐就行了。
```nginx
server {
    listen       80;
    server_name  localhost;

    location / {
        set_by_lua_block $bind_ip {
            return '2001:1111:1111:1111:1111:1111:1111:' .. string.format('%x', math.random(1, 1000))
        }
        proxy_bind $bind_ip;
        proxy_pass http://example.com;
        proxy_set_header Host example.com;
    }
}
```

**Comment by JKme** [comment]
CloudFlare增加IPv6黑名单的时候是以/64段封禁的 <https://developers.cloudflare.com/firewall/cf-dashboard/rules-lists/manage-items#add-items-to-a-list>

**Reply by elizabeth-wilona** [reply]
符合编址规范中的要求,一个/64指的就是一个终端,现在国内有些地区分配/56,大多数地区还是分配/64.

**Comment by xianshang** [comment]
github的代理池可以用来扫描网站吗

**Comment by canugurlu** [comment]
not working...

**Reply by zu1k** [reply]
any detail?

**Reply by canugurlu** [reply]
Thaks for your reply.

$ curl --interface MYIPV6::1 ipv6.ip.sb

I am trying to run this command but waiting, not responding any text.

**Reply by canugurlu** [reply]
And i have tried Ubuntu 18.04 and Debian 10, any advice for OS?

**Reply by zu1k** [reply]
Make sure you do have an entire IPv6 cidr first.

Ubuntu 18 and Debian 10 should work.

**Reply by canugurlu** [reply]
https://prnt.sc/GtMtgFE_o3tl

Stucking here...

**Reply by zu1k** [reply]
If your ndppd is already running, you should restart instead of start.

**Reply by canugurlu** [reply]
I tried, same problem...

**Reply by zu1k** [reply]
have you  tried -vv to see verbose log?

**Reply by canugurlu** [reply]
Oh, I am so stupid, it is worked like a charm on Debian 10 mate!

But I am newbie in programming, how to run rust script for proxy server? Any guide?

**Reply by canugurlu** [reply]
https://prnt.sc/tAHCMI-s3lBI

How to run your proxy script?

Thanks for everything!

**Reply by zu1k** [reply]
Rust is not a script language, you need to compile first. You need to solve these simple problems yourself

**Reply by canugurlu** [reply]
I did it, thanks again. All things working now. How to reach this proxy outside like serverpublicIP:port?

**Reply by zu1k** [reply]
change The bind addr

**Reply by canugurlu** [reply]
How to do that mate, sorry for my stupid questions .. :(

Which step for bind change?

**Reply by zu1k** [reply]
-h to see help
-b 0.0.0.0:5080 to bind all ip

**Comment by yanan-wang** [comment]
请问下大佬是怎么自动切换ipv6地址去访问的？我用
while true; do curl -x http://127.0.0.1:51080 ipv6.ip.sb; done
每次只能得到一个地址。

**Reply by zu1k** [reply]
首先你的确定你有很多 IPv6 地址，然后按照本文的方法配置

**Reply by yanan-wang** [reply]
是按照本文的方法配置的，也是用的vultr服务器，不知道哪里出了问题。

**Reply by zu1k** [reply]
本文中的验证过程，你有验证通过吗？

**Reply by yanan-wang** [reply]
感谢感谢，刚才又试了下，现在可以啦。bind那块出的问题。

**Comment by fangwangme** [comment]
Vultr 不会限制吗？
之前申请提升 limit 的时候，说不支持做爬虫的。
搭代理服务的话，感觉比使用代理的爬虫影响还大

**Reply by zu1k** [reply]
只能说不要滥用

**Comment by yuanweize** [comment]
buyvm可以吗? 分配的是一个Link-local IPv6 Subnet, 但是需要手动分配ipv6,最多一次8个

**Reply by dt3989** [reply]
你找到答案了吗？

**Comment by Who-AmI** [comment]
为何我用curl 测试 发现返回 ‘Empty reply from server’， 但不挂代理就是正常返回的，求解

**Comment by Herts** [comment]
太棒了，我一直没找到ipv6的这个用法。

**Comment by huxiao999** [comment]
是否可以实现用户名密码方式的代理池呢，每个用户对应一个IP，对于每个用户来说，分配的代理IP是静态的

**Reply by zu1k** [reply]
可以实现，不过你得自己改代码

**Reply by zu1k** [reply]
我曾经改过一个 `shadowsocks-rust` 的版本，用户可以用子网下的任意 ip 连接，连接后访问的流量会以用户连接的 ip 为出口访问目标网站，还挺有意思的

**Reply by huxiao999** [reply]
谢谢您的提示，我去看看

**Reply by huxiao999** [reply]
用Squid的ACL就可以很好的实现基于用户密码方式管理的http代理，我觉得这样挺好的

**Reply by HenryWren** [reply]
老哥能提供下ACL  指定账户 =>指定Ipv6的命令吗

**Reply by Yesuki** [reply]
@zulk 坛主，这个版本的github链接方便提供吗？这貌似是代理(Proxy)跟Spoofing二合一了。谢谢！

**Comment by nvtinh368** [comment]
我已经有 1 台 vultr 服务器
我使用 ubuntu 18.4
我的设置和你一样，但是当我执行命令时
curl --interface 2001:19f0:5401:74c::1 http://icanhazip.com/
没啥事儿

**Reply by HenryWren** [reply]
路由是不是没配对

**Reply by zu1k** [reply]
如果各步骤都做对了，那就重启 ndppd 服务

```
sudo systemctl restart ndppd
```

**Reply by HenryWren** [reply]
可是能是ndppd 服务没启动，你ps 查看下呢， 我之前使用 systemctl start ndppd没效果，单独nohup 去把ndppd运行了才可以

**Reply by nvtinh368** [reply]
当我运行 >500 个线程时代理开始失败

**Reply by nvtinh368** [reply]
如何能够同时运行超过 10k 个线程？

**Reply by HenryWren** [reply]
大概率是转发层的问题，带宽，机器性能都要考虑，这个量级的并发，肯定不是几台机器就能解决的

**Reply by nvtinh368** [reply]
使用 1 个 CPU 和 1 GB 服务器，当我同时运行超过 500 个线程时，它会出现错误情况
 println!("error: {addr_str}");

**Reply by nvtinh368** [reply]
我的时候也遇到过这个问题。 使用 300 个线程运行爬网一切正常，关闭然后重新打开使用 300 个线程的爬网程序并抛出错误

**Reply by Yesuki** [reply]
可以联系我，可能我能解决

**Comment by akiyamamio0921** [comment]
openvz 怎么搞 ip a 出来的网卡不是一般的 enp1s0 
```
venet0: <BROADCAST,POINTOPOINT,NOARP,ALLMULTI,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default 
    link/void 
    inet 127.0.0.1/32 scope host venet0
       valid_lft forever preferred_lft forever
    inet 192.168.0.149/22 brd 192.168.3.255 scope global venet0:0
       valid_lft forever preferred_lft forever
    inet6 2001:41d0:1004:335b:1234:1234:2a3a:1/112 scope global 
       valid_lft forever preferred_lft forever
```
ip route
```
root@374758940woiden ~# ip route show
default dev venet0 scope link 
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 
192.168.0.0/22 dev venet0 proto kernel scope link src 192.168.0.149 
root@374758940woiden ~# ip route show default
default dev venet0 scope link 
root@374758940woiden ~# ip route show dev
Command line is not complete. Try option "help"
root@374758940woiden ~ [255]# ip route show dev venet0
default scope link 
192.168.0.0/22 proto kernel scope link src 192.168.0.149 
```
这个 ipv6 是有 65536 个子网的意思吗 
curl测试不成功 连不到服务  我按着你这个来的 openvz 带 tun 的那种有没有教程

**Reply by zu1k** [reply]
我没用过 openvz ，自行探索一下吧

**Comment by saobilin** [comment]
大佬您好
我的服务器提供了一个/64但是他系统模板直接配置了/64的ipv6
iface eth0 inet6 static
    address 2606:挡一下:1:64::a/64
    gateway 2606:挡一下:1::1
我该如何去修改这个设置 目前按照您的教程出口不通

**Reply by zu1k** [reply]
确认一下整个 /64 都是路由到你的主机上，像 vultr 可以直接用 ndproxy，而 linode 需要单独开工单来申请，否则只有一个 ip 地址是路由的

**Reply by saobilin** [reply]
我测试一个网卡绑定了俩ip 路由正常是通的 实在没啥头绪😂能赐教一下吗

**Reply by HenryWren** [reply]
先把系统给你网卡配的ipv6 给清了，然后你自己配一张虚拟网卡，再把 ip route 指到虚拟网卡上 ,v6的 ip_nonlocal_bind记得要开

**Reply by saobilin** [reply]
> 先把系统给你网卡配的ipv6 给清了，然后你自己配一张虚拟网卡，再把 ip route 指到虚拟网卡上 ,v6的 ip_nonlocal_bind记得要开

大佬您好 按照你说的 我配置了 eth0:0的网卡并且只绑定了一个ipv6地址 
ip route add local 2606:挡一下:1:64::/64 dev eth0:0
然后再启动ndppd 还是不可以 如果方便的话能否留个邮箱或者telegram 我给您发ssh您试试 如果成功的话可以请您吃疯狂星期四四

**Comment by gopkg-dev** [comment]
Currently Installed: Debian 11 x64 (bullseye)
```shell
root@vultr:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp1s0: <BROADCAST,MULTICAST,ALLMULTI,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 56:00:04:61:18:3d brd ff:ff:ff:ff:ff:ff
    inet 45.32.16.212/23 brd 45.32.17.255 scope global dynamic enp1s0
       valid_lft 82550sec preferred_lft 82550sec
    inet6 2001:19f0:7001:200a:5400:4ff:fe61:183d/64 scope global dynamic mngtmpaddr 
       valid_lft 2591924sec preferred_lft 604724sec
    inet6 fe80::5400:4ff:fe61:183d/64 scope link 
       valid_lft forever preferred_lft forever
root@vultr:~# ip route add local 2001:19f0:7001:200a::64 dev enp1s0
root@vultr:~# sysctl net.ipv6.ip_nonlocal_bind=1
net.ipv6.ip_nonlocal_bind = 1
root@vultr:~# apt install ndppd
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  ndppd
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 65.1 kB of archives.
After this operation, 197 kB of additional disk space will be used.
Get:1 https://deb.debian.org/debian bullseye/main amd64 ndppd amd64 0.2.5-6 [65.1 kB]
Fetched 65.1 kB in 0s (369 kB/s)
Selecting previously unselected package ndppd.
(Reading database ... 73902 files and directories currently installed.)
Preparing to unpack .../ndppd_0.2.5-6_amd64.deb ...

Progress: [  0%] [..........................................................................................................................................................] 
Unpacking ndppd (0.2.5-6) ...###################............................................................................................................................] 

Setting up ndppd (0.2.5-6) ...#################################################.............................................................................................] 

Progress: [ 60%] [############################################################################################..............................................................] 
Processing triggers for man-db (2.9.4-2) ...#################################################################################################...............................] 
root@vultr:~# systemctl start ndppd
root@vultr:~# curl --interface 2001:19f0:7001:200a::64 ipv6.ip.sb
2001:19f0:7001:200a::64
root@vultr:~# curl --interface 2001:19f0:7001:200a::2 ipv6.ip.sb
curl: (28) Failed to connect to ipv6.ip.sb port 80: Connection timed out
```


会卡住不知道为什么??

**Reply by saobilin** [reply]
ip route你确定写对了？
有无/etc/ndppd.conf？
有，请restart ndppd

**Reply by gopkg-dev** [reply]
可以了,博主这个http-proxy-ipv6-pool是可以当代理用吗?  我尝试在浏览器使用没有用.

**Reply by saobilin** [reply]
使用代理软件 http协议 自己研究 可以用clash

**Reply by gopkg-dev** [reply]
./http-proxy-ipv6-pool -b 0.0.0.0:51080 -i 2001:19f0:7001:200a::/64

http 代理没有用,我用代理工具检测不行,但是在服务器上可以通过

在服务器请求又可以
```
root@vultr:~# curl --interface 2001:19f0:7001:200a::1 ipv6.ip.sb
2001:19f0:7001:200a::1
root@vultr:~#
```

该怎么做让它能代理我浏览器的请求

**Reply by zu1k** [reply]
ip route add 写错了

**Reply by gopkg-dev** [reply]
@zu1k  是的我搞好了,但是如何才能在我本地使用这个代理请求,我尝试用 http proxy的方式不太行.

**Reply by zu1k** [reply]
`while true; do curl -x http://127.0.0.1:51080 ipv6.ip.sb; done` 不行吗？加 -vv  看看什么问题

**Reply by saobilin** [reply]
你vultr防火墙关了吗？端口放行了吗？ip被墙了吗？

**Reply by gopkg-dev** [reply]
@zu1k 在服务器上执行这个是可以的,但是我本机通过 IP:51080连不上,但是 telnet 的通端口就很奇怪,
@saobilin 防火墙没开,能 telnet 通应该没有问题啊

**Reply by pkuntik** [reply]
一样的问题，解决了吗

**Reply by gopkg-dev** [reply]
@zu1k @saobilin 两位大佬你们的可以在你们本地使用   http://ServerIP:51080  代理吗?

**Reply by saobilin** [reply]
> @zu1k @saobilin 两位大佬你们的可以在你们本地使用 http://ServerIP:51080 代理吗?
你找台国外机器装windows试试得了

**Reply by pkuntik** [reply]
解决了，装了个宝塔 用宝塔放行端口

**Reply by gopkg-dev** [reply]
vultr 账号被封了

**Reply by zu1k** [reply]
> vultr 账号被封了

用来做什么了？

**Reply by gopkg-dev** [reply]
> > vultr 账号被封了
> 
> 用来做什么了？

他说我开代理了

**Reply by HenryWren** [reply]
自己租ip 广播就没事

**Comment by pkuntik** [comment]
阿里云是不是不行

**Reply by zu1k** [reply]
不清楚，你可以自己尝试一下，或者咨询阿里云客服，我没有阿里云的服务器

**Reply by gopkg-dev** [reply]
阿里云测试可以

**Reply by zu1k** [reply]
> 阿里云测试可以

感谢反馈。我看到你之前提到 vultr 帐号因为这个被封了，还是要注意一下不要滥用

**Reply by starxg** [reply]
阿里云测试成功了吗？试了一下好像不行

**Reply by gopkg-dev** [reply]
TG UUd0aGNtVnVYMkp2ZVElM0QlM0Q=

**Comment by knva** [comment]
很奇怪，手动配置一个IP 然后使用curl --interface 可以访问
使用route +ndppd +sysctl net.ipv6.ip_nonlocal_bind=1 无法访问

**Reply by knva** [reply]
curl --interface 2604:a00:50:2d::233 6.ipw.cn -vv
*   Trying 2402:4e00:1013:e500:0:9671:f018:4947:80...
* Name '2604:a00:50:2d::233' family 10 resolved to '2604:a00:50:2d::233' family 10
* Local port: 0
卡住

**Reply by knva** [reply]
已解决  
```bash
ip route add local 2604:a00:50:2d::/64 dev lo 
```

**Reply by knva** [reply]
使用的是he.net的隧道

**Reply by knva** [reply]
怀疑是ndppd的问题。因为he.net不需要这个就可以运行

**Reply by zu1k** [reply]
he.net 直接路由过来了，确实不需要 ndp 代理

**Comment by gopkg-dev** [comment]
你们的能访问谷歌吗?

# 服务器上正常返回

```
curl -x http://127.0.0.1:5648 -vv https://www.google.com.hk
* Rebuilt URL to: https://www.google.com.hk/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 5648 (#0)
* allocate connect buffer!
* Establish HTTP proxy tunnel to www.google.com.hk:443
> CONNECT www.google.com.hk:443 HTTP/1.1
> Host: www.google.com.hk:443
> User-Agent: curl/7.58.0
> Proxy-Connection: Keep-Alive
> 
< HTTP/1.1 200 OK
< Date: Wed, 07 Jun 2023 05:55:39 GMT
< 
* Proxy replied 200 to CONNECT request
* CONNECT phase completed!
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* CONNECT phase completed!
* CONNECT phase completed!
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Unknown (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Client hello (1):
* TLSv1.3 (OUT), TLS Unknown, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.google.com.hk
*  start date: May 19 12:59:11 2023 GMT
*  expire date: Aug 11 12:59:10 2023 GMT
*  subjectAltName: host "www.google.com.hk" matched cert's "*.google.com.hk"
*  issuer: C=US; O=Google Trust Services LLC; CN=GTS CA 1C3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* Using Stream ID: 1 (easy handle 0x563b818e8640)
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
> GET / HTTP/2
> Host: www.google.com.hk
> User-Agent: curl/7.58.0
> Accept: */*
> 
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
< HTTP/2 200 
< date: Wed, 07 Jun 2023 05:55:39 GMT
< expires: -1
< cache-control: private, max-age=0
< content-type: text/html; charset=Big5
< content-security-policy-report-only: object-src 'none';base-uri 'self';script-src 'nonce-C1s_K87iyRuARg-pP3LGfQ' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp
< p3p: CP="This is not a P3P policy! See g.co/p3phelp for more info."
< server: gws
< x-xss-protection: 0
< x-frame-options: SAMEORIGIN
< set-cookie: 1P_JAR=2023-06-07-05; expires=Fri, 07-Jul-2023 05:55:39 GMT; path=/; domain=.google.com.hk; Secure
< set-cookie: AEC=AUEFqZdOS-_BpCcHr1cT-w_kTYXNeG3RduxdlR0yO3Np5t21oK-bHFsYGy4; expires=Mon, 04-Dec-2023 05:55:39 GMT; path=/; domain=.google.com.hk; Secure; HttpOnly; SameSite=lax
< set-cookie: NID=511=r_bSCV-6weMgmj7FrX-SqxOBJ4seXa4jgzvOdBYfdIoq9mgx1g_T2wP9nof9HrvFTiDipTG0wUvQvyfC_8bgPEHZOuKG2l8YwUL_Nep4MmcjJVYBC8y0kGFvGmjYyuFiR9tO5i4ziOKWLQmzPqs_DJiq3cL6Sw9D-1ZbsFuLDKw; expires=Thu, 07-Dec-2023 05:55:39 GMT; path=/; domain=.google.com.hk; HttpOnly
< alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
< accept-ranges: none
< vary: Accept-Encoding
< 
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
<!doctype html> ...........
```

# 本机不行
```
curl -x http://<IP>:5648 -vv https://www.google.com.hk
*   Trying <IP>...
* Connected to <IP> (<IP>) port 5648 (#0)
* allocate connect buffer
* Establish HTTP proxy tunnel to www.google.com.hk:443
> CONNECT www.google.com.hk:443 HTTP/1.1
> Host: www.google.com.hk:443
> User-Agent: curl/7.88.1
> Proxy-Connection: Keep-Alive
>
* Proxy CONNECT aborted
* Closing connection 0
curl: (56) Proxy CONNECT aborted
```

好像因为连接没成功就返回了? 还是被gfw给墙了? 能解决吗?

**Reply by HenryWren** [reply]
被墙了

**Comment by gopkg-dev** [comment]
我用Puppeteer使用代理,会出现net::ERR_CONNECTION_RESET,好像是connect的问题.

**Comment by yunzheng2006** [comment]
其实 v6 并没有成本，在 RIPE NCC 这样的 RIR 可以不需要任何理由拿一个 /29 的几辈子用不完的 v6 地址段，（前提你得跟我一样是 RIPE NCC 会员）；或者你可以去一些网站自己租用 /48 /44 等等 v6地址段 自己通过一些服务商广播，成本更低；

**Reply by HenryWren** [reply]
LIR 成本高啊，门槛就1W起了，除非商用，少量的话性价比最高的是/40的，有一家我记得是10$一年

**Reply by Yesuki** [reply]
方便提供/40哪里有提供吗？谢谢！

**Reply by yunzheng2006** [reply]
Telegram @ash_59538

**Reply by Yesuki** [reply]
有其它的联系方式提供吗？我的Telegram账号现在异常。谢谢！

**Comment by on** [comment]
Warning: Ignores instruction to use SSLv3
*   Trying [XXXX:XXXX::XXXX:XXXX]:80...
* Name 'YYYY:YYYY:YYYY:YYY::1' family 10 resolved to 'YYYY:YYYY:YYYY:YYY::1' family 10
* Local port: 0
* connect to XXXX:XXXX::XXXX:XXXX port 80 failed: Connection timed out
*   Trying [XXXX:XXXX::XXXX:XXXX]:80...
* Hostname in DNS cache was stale, zapped
* Name 'YYYY:YYYY:YYYY:YYY::1' family 10 resolved to 'YYYY:YYYY:YYYY:YYY::1' family 10
* Local port: 0

any help?

**Comment by zhuxindong** [comment]
能不能转换为socks5的代理？http代理的限制太多了

**Reply by zu1k** [reply]
实现起来应该很容易，你可以尝试二次开发一下

**Comment by Rainscall** [comment]
使用ip6tables实现了伪随机
机器上大部分ipv6出口都可以走随机了
性能应该也会高一点
https://www.typeboom.com/archives/112/

**Comment by Fangliding** [comment]
哈哈 感谢大佬的文章
我在xray上实现了一个
https://github.com/XTLS/Xray-core/releases/tag/v1.8.10

**Comment by q629988171** [comment]
局域网的IPv4能用这个方法吗？起因是我忘记路由器的密码又不想重置，打算用穷举法不断的尝试登录路由器的后台，但是这个后台不仅有验证码而且连续错5次就锁IP，我就得换个IP再试，手动换IP太影响穷举的速度。

**Reply by zu1k** [reply]
可以的

**Comment by Tsumeta** [comment]
你好 可以有偿搭建吗

**Comment by justruiqwq** [comment]
如果自己有asn或者找服务商代播成本很低了，一段/40价格大概10刀左右

**Reply by starxg** [reply]
怎么操作呢？

**Reply by gopkg-dev** [reply]
我是自己$12 买了个/40 的然后租了台支持BGP Session 的机器, 借的朋友的ASN ， 可以自己找个靠谱的LIR services 自己注册 asn
